lib/emerald/parser.rb

```diff
@@ -27,9 +37,15 @@ def parse_node(source)
       when /[\d]/
         parse_number(source)
       when /[+-]/
```

Comment:
How about `when /[+-]?\d/` for numbers, then we don't need the `if` nested inside to differentiate between numbers and atoms.